In [2]:
import pandas as pd



Using the datasets above: List_of_local_governments_in_Nigeria.csv and Nigeria_povertyrate_per_local_gov.csv
    1. Merge the dataset together to form a single table.
    2. Remove any redundant columns.
    3. Calculate how population affects poverty rate of each local government.
    4. Arrange each local government accordiing to the order of their poverty rate.
    5. Find the percentage of poverty and sort in ascending order.

In [3]:
local_governments = pd.read_csv("list_of_local_government_areas_of_nigeria.csv")
local_governments

SNo Local Government Area  \
0                                                    1                Abadam   
1                                                    2                 Abaji   
2                                                    3                  Abak   
3                                                    4             Abakaliki   
4                                                    5             Aba North   
..                                                 ...                   ...   
771                                                772                  Zing   
772                                                773                 Zurmi   
773                                                774                  Zuru   
774                                                NaN                   NaN   
775  For more information, kindly visit :  www.down...                   NaN   

               State  Unnamed: 3  
0        Borno State         NaN  
1                FCT         NaN  
2    Akwa Ibom State         NaN  
3       Ebonyi State         NaN  
4         Abia State         NaN  
..               ...         ...  
771     Taraba State         NaN  
772    Zamfara State         NaN  
773      Kebbi State         NaN  
774              NaN         NaN  
775              NaN         NaN  

[776 rows x 4 columns]

In [4]:
local_governments.isnull().any()

SNo                      True
Local Government Area    True
State                    True
Unnamed: 3               True
dtype: bool

In [5]:
lga = local_governments.drop(columns = "Unnamed: 3")

In [6]:
lga

SNo Local Government Area  \
0                                                    1                Abadam   
1                                                    2                 Abaji   
2                                                    3                  Abak   
3                                                    4             Abakaliki   
4                                                    5             Aba North   
..                                                 ...                   ...   
771                                                772                  Zing   
772                                                773                 Zurmi   
773                                                774                  Zuru   
774                                                NaN                   NaN   
775  For more information, kindly visit :  www.down...                   NaN   

               State  
0        Borno State  
1                FCT  
2    Akwa Ibom State  
3       Ebonyi State  
4         Abia State  
..               ...  
771     Taraba State  
772    Zamfara State  
773      Kebbi State  
774              NaN  
775              NaN  

[776 rows x 3 columns]

In [7]:
poverty = pd.read_csv("Nigeria povertyrate per local gov.csv")
poverty.head()

ID_0  ISO   NAME_0  ID_12 NAME_12  ID_23    NAME_23  2014_Popul  \
0   163  NGA  Nigeria      1    Abia      1  Aba North      257689   
1   163  NGA  Nigeria      1    Abia      2  Aba South      405791   
2   163  NGA  Nigeria      1    Abia      3   Arochukw      193250   
3   163  NGA  Nigeria      1    Abia      4      Bende      244399   
4   163  NGA  Nigeria      1    Abia      5    Ikwuano      139157   

   Poverty_Ra  2014_Pover  Unnamed: 10  
0    0.017378        9661          NaN  
1    0.094480       16070          NaN  
2    0.175241       40022          NaN  
3    0.129469       39410          NaN  
4    0.111351       12145          NaN

In [8]:
pov = poverty.drop(columns = "Unnamed: 10")
pov

ID_0  ISO   NAME_0  ID_12  NAME_12  ID_23        NAME_23  2014_Popul  \
0     163  NGA  Nigeria      1     Abia      1      Aba North      257689   
1     163  NGA  Nigeria      1     Abia      2      Aba South      405791   
2     163  NGA  Nigeria      1     Abia      3       Arochukw      193250   
3     163  NGA  Nigeria      1     Abia      4          Bende      244399   
4     163  NGA  Nigeria      1     Abia      5        Ikwuano      139157   
..    ...  ...      ...    ...      ...    ...            ...         ...   
770   163  NGA  Nigeria     38  Zamfara    771           Maru      449533   
771   163  NGA  Nigeria     38  Zamfara    772       Shinkafi      220070   
772   163  NGA  Nigeria     38  Zamfara    773  Talata-Mafara      131298   
773   163  NGA  Nigeria     38  Zamfara    774          Tsafe      333706   
774   163  NGA  Nigeria     38  Zamfara    775          Zurmi      291140   

     Poverty_Ra  2014_Pover  
0      0.017378        9661  
1      0.094480       16070  
2      0.175241       40022  
3      0.129469       39410  
4      0.111351       12145  
..          ...         ...  
770    0.502184      227452  
771    0.390775       84238  
772    0.554036       72134  
773    0.565074      186716  
774    0.486278      144487  

[775 rows x 10 columns]

In [9]:
pov.describe()

ID_0       ID_12      ID_23    2014_Popul  Poverty_Ra     2014_Pover
count  775.0  775.000000  775.00000  7.750000e+02  775.000000     775.000000
mean   163.0   19.237419  388.00000  2.300360e+05    0.324967   62064.329032
std      0.0   10.558213  223.86752  1.519788e+05    0.187816   48482.699459
min    163.0    1.000000    1.00000  1.998000e+03    0.004566      86.000000
25%    163.0   10.000000  194.50000  1.401310e+05    0.167397   24859.500000
50%    163.0   20.000000  388.00000  1.961350e+05    0.315857   49679.000000
75%    163.0   28.000000  581.50000  2.712220e+05    0.464558   88379.500000
max    163.0   38.000000  775.00000  1.401650e+06    0.914384  238909.000000

In [10]:
new_data=pd.merge(lga, pov, left_on ="Local Government Area", right_on ="NAME_23")
new_data

SNo Local Government Area            State  ID_0  ISO   NAME_0  ID_12  \
0      1                Abadam      Borno State   163  NGA  Nigeria      8   
1      2                 Abaji              FCT   163  NGA  Nigeria     15   
2      3                  Abak  Akwa Ibom State   163  NGA  Nigeria      3   
3      5             Aba North       Abia State   163  NGA  Nigeria      1   
4      6             Aba South       Abia State   163  NGA  Nigeria      1   
..   ...                   ...              ...   ...  ...      ...    ...   
590  769                 Zango    Katsina State   163  NGA  Nigeria     21   
591  771                 Zaria     Kaduna State   163  NGA  Nigeria     19   
592  772                  Zing     Taraba State   163  NGA  Nigeria     35   
593  773                 Zurmi    Zamfara State   163  NGA  Nigeria     38   
594  774                  Zuru      Kebbi State   163  NGA  Nigeria     22   

                       NAME_12  ID_23    NAME_23  2014_Popul  Poverty_Ra  \
0                        Borno    142     Abadam       83113    0.396480   
1    Federal Capital Territory    276      Abaji       60985    0.243414   
2                    Akwa Ibom     39       Abak      153585    0.227546   
3                         Abia      1  Aba North      257689    0.017378   
4                         Abia      2  Aba South      405791    0.094480   
..                         ...    ...        ...         ...         ...   
590                    Katsina    447      Zango      217949    0.558352   
591                     Kaduna    369      Zaria      448419    0.122632   
592                     Taraba    743       Zing      113311    0.494821   
593                    Zamfara    775      Zurmi      291140    0.486278   
594                      Kebbi    468       Zuru      159716    0.362035   

     2014_Pover  
0         31093  
1         14320  
2         33179  
3          9661  
4         16070  
..          ...  
590      117828  
591       21111  
592       56127  
593      144487  
594       51528  

[595 rows x 13 columns]

In [11]:
car = new_data.drop(columns = ["NAME_12", "NAME_23"])
car

SNo Local Government Area            State  ID_0  ISO   NAME_0  ID_12  \
0      1                Abadam      Borno State   163  NGA  Nigeria      8   
1      2                 Abaji              FCT   163  NGA  Nigeria     15   
2      3                  Abak  Akwa Ibom State   163  NGA  Nigeria      3   
3      5             Aba North       Abia State   163  NGA  Nigeria      1   
4      6             Aba South       Abia State   163  NGA  Nigeria      1   
..   ...                   ...              ...   ...  ...      ...    ...   
590  769                 Zango    Katsina State   163  NGA  Nigeria     21   
591  771                 Zaria     Kaduna State   163  NGA  Nigeria     19   
592  772                  Zing     Taraba State   163  NGA  Nigeria     35   
593  773                 Zurmi    Zamfara State   163  NGA  Nigeria     38   
594  774                  Zuru      Kebbi State   163  NGA  Nigeria     22   

     ID_23  2014_Popul  Poverty_Ra  2014_Pover  
0      142       83113    0.396480       31093  
1      276       60985    0.243414       14320  
2       39      153585    0.227546       33179  
3        1      257689    0.017378        9661  
4        2      405791    0.094480       16070  
..     ...         ...         ...         ...  
590    447      217949    0.558352      117828  
591    369      448419    0.122632       21111  
592    743      113311    0.494821       56127  
593    775      291140    0.486278      144487  
594    468      159716    0.362035       51528  

[595 rows x 11 columns]

In [12]:
new =car.set_index("SNo")
new

Local Government Area            State  ID_0  ISO   NAME_0  ID_12  ID_23  \
SNo                                                                            
1                  Abadam      Borno State   163  NGA  Nigeria      8    142   
2                   Abaji              FCT   163  NGA  Nigeria     15    276   
3                    Abak  Akwa Ibom State   163  NGA  Nigeria      3     39   
5               Aba North       Abia State   163  NGA  Nigeria      1      1   
6               Aba South       Abia State   163  NGA  Nigeria      1      2   
..                    ...              ...   ...  ...      ...    ...    ...   
769                 Zango    Katsina State   163  NGA  Nigeria     21    447   
771                 Zaria     Kaduna State   163  NGA  Nigeria     19    369   
772                  Zing     Taraba State   163  NGA  Nigeria     35    743   
773                 Zurmi    Zamfara State   163  NGA  Nigeria     38    775   
774                  Zuru      Kebbi State   163  NGA  Nigeria     22    468   

     2014_Popul  Poverty_Ra  2014_Pover  
SNo                                      
1         83113    0.396480       31093  
2         60985    0.243414       14320  
3        153585    0.227546       33179  
5        257689    0.017378        9661  
6        405791    0.094480       16070  
..          ...         ...         ...  
769      217949    0.558352      117828  
771      448419    0.122632       21111  
772      113311    0.494821       56127  
773      291140    0.486278      144487  
774      159716    0.362035       51528  

[595 rows x 10 columns]

In [13]:
correla = new.corr(method= "pearson")
correla

ID_0     ID_12     ID_23  2014_Popul  Poverty_Ra  2014_Pover
ID_0         NaN       NaN       NaN         NaN         NaN         NaN
ID_12        NaN  1.000000  0.997631    0.032983   -0.011638   -0.003003
ID_23        NaN  0.997631  1.000000    0.028697   -0.017926   -0.013306
2014_Popul   NaN  0.032983  0.028697    1.000000   -0.173417    0.292340
Poverty_Ra   NaN -0.011638 -0.017926   -0.173417    1.000000    0.755667
2014_Pover   NaN -0.003003 -0.013306    0.292340    0.755667    1.000000

# Although this is an inverse relationship, showing that when population increases, poverty rate decreases but considerin the value obtained (-0.173417), population cannot be used to determine poverty rate

In [14]:
y = new['Poverty_Ra']

In [15]:
import statsmodels.api as sm

In [16]:
x = new['2014_Popul']

In [17]:
X = sm.add_constant(x)

In [18]:
regress =model = sm.OLS(y, X).fit()

In [19]:
regress.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:             Poverty_Ra   R-squared:                       0.030
Model:                            OLS   Adj. R-squared:                  0.028
Method:                 Least Squares   F-statistic:                     18.39
Date:                Tue, 29 Aug 2023   Prob (F-statistic):           2.10e-05
Time:                        21:41:23   Log-Likelihood:                 163.68
No. Observations:                 595   AIC:                            -323.4
Df Residuals:                     593   BIC:                            -314.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.3884      0.013     28.993      0.000       0.362       0.415
2014_Popul -2.114e-07   4.93e-08     -4.288      0.000   -3.08e-07   -1.15e-07
==============================================================================
Omnibus:                       21.345   Durbin-Watson:                   1.700
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               15.212
Skew:                           0.280   Prob(JB):                     0.000498
Kurtosis:                       2.452   Cond. No.                     4.82e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.82e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [33]:
new["Percentage %"]= new["Poverty_Ra"] *100

In [34]:
new

State  ID_0  ISO   NAME_0  ID_12  ID_23  \
Local Government Area                                                      
Abadam                     Borno State   163  NGA  Nigeria      8    142   
Abaji                              FCT   163  NGA  Nigeria     15    276   
Abak                   Akwa Ibom State   163  NGA  Nigeria      3     39   
Aba North                   Abia State   163  NGA  Nigeria      1      1   
Aba South                   Abia State   163  NGA  Nigeria      1      2   
...                                ...   ...  ...      ...    ...    ...   
Zango                    Katsina State   163  NGA  Nigeria     21    447   
Zaria                     Kaduna State   163  NGA  Nigeria     19    369   
Zing                      Taraba State   163  NGA  Nigeria     35    743   
Zurmi                    Zamfara State   163  NGA  Nigeria     38    775   
Zuru                       Kebbi State   163  NGA  Nigeria     22    468   

                       2014_Popul  Poverty_Ra  2014_Pover  percentage  \
Local Government Area                                                   
Abadam                      83113    0.396480       31093    39.64800   
Abaji                       60985    0.243414       14320    24.34140   
Abak                       153585    0.227546       33179    22.75460   
Aba North                  257689    0.017378        9661     1.73775   
Aba South                  405791    0.094480       16070     9.44801   
...                           ...         ...         ...         ...   
Zango                      217949    0.558352      117828    55.83520   
Zaria                      448419    0.122632       21111    12.26320   
Zing                       113311    0.494821       56127    49.48210   
Zurmi                      291140    0.486278      144487    48.62780   
Zuru                       159716    0.362035       51528    36.20350   

                       Percentage  Percentage %  
Local Government Area                            
Abadam                   39.64800      39.64800  
Abaji                    24.34140      24.34140  
Abak                     22.75460      22.75460  
Aba North                 1.73775       1.73775  
Aba South                 9.44801       9.44801  
...                           ...           ...  
Zango                    55.83520      55.83520  
Zaria                    12.26320      12.26320  
Zing                     49.48210      49.48210  
Zurmi                    48.62780      48.62780  
Zuru                     36.20350      36.20350  

[595 rows x 12 columns]

In [28]:
indData = new.set_index("Local Government Area", inplace=True)

In [35]:
new

State  ID_0  ISO   NAME_0  ID_12  ID_23  \
Local Government Area                                                      
Abadam                     Borno State   163  NGA  Nigeria      8    142   
Abaji                              FCT   163  NGA  Nigeria     15    276   
Abak                   Akwa Ibom State   163  NGA  Nigeria      3     39   
Aba North                   Abia State   163  NGA  Nigeria      1      1   
Aba South                   Abia State   163  NGA  Nigeria      1      2   
...                                ...   ...  ...      ...    ...    ...   
Zango                    Katsina State   163  NGA  Nigeria     21    447   
Zaria                     Kaduna State   163  NGA  Nigeria     19    369   
Zing                      Taraba State   163  NGA  Nigeria     35    743   
Zurmi                    Zamfara State   163  NGA  Nigeria     38    775   
Zuru                       Kebbi State   163  NGA  Nigeria     22    468   

                       2014_Popul  Poverty_Ra  2014_Pover  percentage  \
Local Government Area                                                   
Abadam                      83113    0.396480       31093    39.64800   
Abaji                       60985    0.243414       14320    24.34140   
Abak                       153585    0.227546       33179    22.75460   
Aba North                  257689    0.017378        9661     1.73775   
Aba South                  405791    0.094480       16070     9.44801   
...                           ...         ...         ...         ...   
Zango                      217949    0.558352      117828    55.83520   
Zaria                      448419    0.122632       21111    12.26320   
Zing                       113311    0.494821       56127    49.48210   
Zurmi                      291140    0.486278      144487    48.62780   
Zuru                       159716    0.362035       51528    36.20350   

                       Percentage  Percentage %  
Local Government Area                            
Abadam                   39.64800      39.64800  
Abaji                    24.34140      24.34140  
Abak                     22.75460      22.75460  
Aba North                 1.73775       1.73775  
Aba South                 9.44801       9.44801  
...                           ...           ...  
Zango                    55.83520      55.83520  
Zaria                    12.26320      12.26320  
Zing                     49.48210      49.48210  
Zurmi                    48.62780      48.62780  
Zuru                     36.20350      36.20350  

[595 rows x 12 columns]

In [46]:
sorted_df = new.sort_values(by="Percentage %", ascending=True, inplace=True)

In [52]:
new.drop(columns =["Percentage", "percentage"])

State  ID_0  ISO   NAME_0  ID_12  ID_23  \
Local Government Area                                                   
Fagge                    Kano State   163  NGA  Nigeria     20    381   
Sokoto North           Sokoto State   163  NGA  Nigeria     34    720   
Uvwie                   Delta State   163  NGA  Nigeria     10    208   
Aba North                Abia State   163  NGA  Nigeria      1      1   
Bida                    Niger State   163  NGA  Nigeria     27    541   
...                             ...   ...  ...      ...    ...    ...   
Ohaozara               Ebonyi State   163  NGA  Nigeria     11    222   
Ezza North             Ebonyi State   163  NGA  Nigeria     11    216   
Ohaukwu                Ebonyi State   163  NGA  Nigeria     11    223   
Ezza South             Ebonyi State   163  NGA  Nigeria     11    217   
Ikwo                   Ebonyi State   163  NGA  Nigeria     11    218   

                       2014_Popul  Poverty_Ra  2014_Pover  Percentage %  
Local Government Area                                                    
Fagge                      859163    0.004566        4250       0.45664  
Sokoto North               246991    0.009919        1496       0.99191  
Uvwie                      289312    0.016641        4621       1.66406  
Aba North                  257689    0.017378        9661       1.73775  
Bida                       240860    0.018769        5264       1.87686  
...                           ...         ...         ...           ...  
Ohaozara                   167245    0.842780      140108      84.27800  
Ezza North                 158054    0.844616      131690      84.46160  
Ohaukwu                    253736    0.862090      220785      86.20900  
Ezza South                 168892    0.903956      151608      90.39560  
Ikwo                       215257    0.914384      197512      91.43840  

[595 rows x 10 columns]